### PR TITLE
Implements FilterMany operator #58

### PR DESCRIPTION
**Pull request by HugoSay**
Implements FilterMany as presented in  #58

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -440,17 +441,38 @@ combineLatest: [false, true, true, true]
...
+.sink(receiveValue: { print($0) })
+
+intArrayPublisher.send([10, 1, 2, 4, 3, 8])
+```
+
+#### Output:
+
+```
+none
+[10, 1, 2, 4, 8]
```
Comment (kevinrenskers) [inline README.md:460]: That 1 shouldn't be there in the example's output I think?

Comment (freak4pc) [reply README.md:460]: yup seems like this should be 10 , 2, 4 , 8 , right ?

File: README.md [hunk 2]
```diff
@@ -440,17 +441,38 @@ combineLatest: [false, true, true, true]
...
 ```swift
 let intArrayPublisher = PassthroughSubject<[Int], Never>()
-    
+
 intArrayPublisher
-  .mapMany(String.init)
-  .sink(receiveValue: { print($0) })
-    
+.mapMany(String.init)
+.sink(receiveValue: { print($0) })
```
Comment (freak4pc) [inline README.md:474]: undo

File: Sources/Operators/FilterMany.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
+//
+//  MapMany.swift
+//  CombineExt
+//
+//  Created by Joan Disho on 22/03/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
```
Comment (freak4pc) [inline Sources/Operators/FilterMany.swift:7]:
```suggestion
//
//  FilterMany.swift
//  CombineExt
//
//  Created by YOUR NAME on 22/03/2020.
//  Copyright © 2020 Combine Community. All rights reserved.
//
```

File: Sources/Operators/FilterMany.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	///
+	///    intArrayPublisher.send([10, 1, 2, 4, 3, 8])
+	///
+	///    // Output: [10, 1, 2, 4, 8]
+	///    ```
+	///
+	///
+	func filterMany(_ isIncluded: @escaping (Output.Element) -> Bool) -> Publishers.Map<Self, [Output.Element]> {
+		map { $0.filter(isIncluded) }
+	}
```
Comment (freak4pc) [inline Sources/Operators/FilterMany.swift:37]: I think the `Map` publisher here doesn't provide a good hint like in the `mapMany` example. I'd say just erase it:
```suggestion
	func filterMany(_ isIncluded: @escaping (Output.Element) -> Bool) -> AnyPublisher<[Output.Element], Output.Failure> {
		map { $0.filter(isIncluded) }.eraseToAnyPublisher()
	}
```

File: Tests/FilterManyTests.swift [hunk 5]
```diff
@@ -0,0 +1,95 @@
+//
+//  MapManyTests.swift
+//  CombineExtTests
+//
+//  Created by Joan Disho on 22/03/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
```
Comment (freak4pc) [inline Tests/FilterManyTests.swift:7]: copyrights

File: Tests/FilterManyTests.swift [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+#if !os(watchOS)
+import XCTest
+import Combine
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+class FilterManyTests: XCTestCase {
+    var subscription: AnyCancellable!
+
+    func testMapManyWithModelAndFinishedCompletion() {
```
Comment (freak4pc) [inline Tests/FilterManyTests.swift:17]:
```suggestion
    func testFilterManyWithModelAndFinishedCompletion() {
```

File: Tests/FilterManyTests.swift [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        source.send(completion: .finished)
+
+        XCTAssertEqual(
+            expectedOutput,
+            [
+                10,
+                2,
+                4,
+                8
+            ]
```
Comment (freak4pc) [inline Tests/FilterManyTests.swift:41]:
```suggestion
            [10, 2, 4, 8]
```

Comment (freak4pc) [reply Tests/FilterManyTests.swift:41]: one line every where else as well

File: Tests/FilterManyTests.swift [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+private extension FilterManyTests {
+    enum MapManyError: Error {
+        case anErrorCase
+    }
+
+    struct SomeModel: Equatable, CustomStringConvertible {
+        let number: Int
+        var description: String { return "#\(number)" }
```
Comment (freak4pc) [inline Tests/FilterManyTests.swift:88]: is description used anywhere ?

**Review by freak4pc — COMMENTED** [review]
thank you! have a few notes

File: README.md (hunk 1 above)
Comment (HugoSay) [reply README.md:460]: 🙈

Comment (HugoSay) [reply README.md]: Fixed

**Comment by HugoSay** [comment]
Pushed the new commit with the corrections. Sorry for the mistakes of my first commit and thanks for your feedbacks!

File: Tests/FilterManyTests.swift (hunk 8 above)
Comment (HugoSay) [reply Tests/FilterManyTests.swift]: Not used indeed, I removed it

**Review by freak4pc — COMMENTED** [review]
looks good - just one holy-war nitpick
this entire repo uses Spaces and you use Tabs.

It's worse because GitHub represents tabs as 8 spaces which makes everything wonky. Could you change to spaces please?

**Comment by HugoSay** [comment]
@freak4pc updated the pr to use spaces instead of tabs 👍.
it's kind of sad that Xcode doesn't let us choose to use spaces or tab on a project basis 🤷‍♂️

**Review by freak4pc — APPROVED** [review]

**Comment by freak4pc** [comment]
Thank you!

**Pull request merged**